src/ui/rcb-ui-inner.svelte

```diff
@@ -36,6 +36,14 @@
   export let purchaseOperationHelper: PurchaseOperationHelper;
   export let isInElement: boolean = false;
 
+  const shouldShowDetailsButton =
+    (Boolean(brandingInfo?.appearance) &&
+      Boolean(productDetails?.description)) ||
+    Boolean(
+      (purchaseOptionToUse as SubscriptionOption)?.trial?.periodDuration,
+    ) ||
+    false;
```

Comment:
I usually prefer to split those complex conditions to make them more readable.
This is an opinionated approach so happy to not use it if you don't agree but I would split it as follows

```suggestion
  const shouldShowProductDescription= (Boolean(brandingInfo?.appearance) &&
      Boolean(productDetails?.description));
  
  const hasTrialInformation= Boolean(
      (purchaseOptionToUse as SubscriptionOption)?.trial?.periodDuration,
    )
  
  const shouldShowDetailsButton = shouldShowProductDescription || hasTrialInformation;
```

Reply:
yeah totally, that was my comment on refactoring, I think these "hasTrialInformation" "showShowProductDescription" etc should probably be in a context, but I'll change it here right now

Reply:
done